tests/middleware/test_message_state.py

```diff
@@ -128,7 +130,17 @@ def do_work(x):
         msg = do_work.send(long_string)
         args = state_backend.get_state(msg.message_id).args
         # if the max_size == 0, then should not storage nothing
+
         if max_size > 200:
             assert list(args) == [long_string]
         else:
             assert args is None
+
+    def test_save_group_id_in_message(self, stub_broker, state_middleware, do_work):
+        msg = do_work.message()
+        group_id = group([msg]).run().group_id
+        state_middleware.after_enqueue = MagicMock()
```

Comment:
Could you remove this line and the line 146 ?

These are false: to test if a mock is called you have to use call_count for example. Here after_enqueue is not called because is was called the line before (`group([msg]).run()`).

When you set a mock, it replace the function behavior and basically do nothing. so if you put the mock before `group([msg]).run()` the state.group_id will not be set.